### PR TITLE
Config docs application connector

### LINKED ISSUE #4052: Create configuration documents for Application Connector
**Description**

Create configuration documents for the **Application Connector** component. 
These documents should list and describe configurable parameters from `values.yaml` files of these charts and their sub-charts:

- [`application-connector-helper`](https://github.com/kyma-project/kyma/tree/master/resources/application-connector-helper)
- [`application-connector-ingress`](https://github.com/kyma-project/kyma/tree/master/resources/application-connector-ingress)
- [`application-connector`](https://github.com/kyma-project/kyma/tree/master/resources/application-connector)

>**NOTE:** Every chart and sub-chart should be described in a separate document.

Make sure you:
- Create the documents based on the existing [template](https://github.com/kyma-project/community/blob/master/guidelines/templates/resources/configuration.md).
- Rely on the already created configuration documents for the [Asset Store](https://kyma-project.io/docs/master/components/asset-store/#configuration-configuration).

**Reasons**

Have a set of configuration documents that describe values from the Application Connector `values.yaml` files you can override.

**Attachments**
It relates to the epic #4028.

### PR DESCRIPTION
**Pull request by mmitoraj**
**Description**

Changes proposed in this pull request:

- Add configuration documents for the Application Connector

**Related issue(s)**

Resolves #4052

### CODE REVIEW

File: docs/application-connector/05-04-connector-service.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## Configurable parameters
+
+This table lists the configurable parameters, their descriptions, and default values:
+
+| Parameter | Description | Default value |
+|-----------|-------------|---------------|
+| **deployment.args.tokenLength**| Specifies a number of characters of a registration token. | `64` |
+| **deployment.args.appTokenExpirationMinutes** | Specifies a period of time after which a token for an Application expires. It is provided in minutes. | `5` |
+| **deployment.args.runtimeTokenExpirationMinutes** | Specifies a period of time after which a token for a runtime expires. It is provided in minutes. | `10` |
+| **deployment.args.appValidityTime** | Specifies a period of time during which certificates that the service issues for an Application are valid. It is provided in days. | `90` |
```
Comment (akgalwas) [inline docs/application-connector/05-04-connector-service.md]: The value contains the unit e.g. '90d' for 90 days or '90h' for 90 hours or '90m' for 90 minutes.

File: docs/application-connector/05-02-application-registry.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+>* [Helm overrides for Kyma installation](/root/kyma/#configuration-helm-overrides-for-kyma-installation)
+>* [Sub-charts overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation-sub-chart-overrides)
+
+## Configurable parameters
+
+This table lists the configurable parameters, their descriptions, and default values:
+
+| Parameter | Description | Default value |
+|-----------|-------------|---------------|
+| **deployment.args.requestTimeout** | Specifies a period of time after which a request to the Application Registry fails to be sent. It is provided in seconds. | `10` |
```
Comment (akgalwas) [inline docs/application-connector/05-02-application-registry.md]: The setting allow to specify what is the summary threshold for request handling time. I think the term "time-out" is more suitable.

File: docs/application-connector/05-04-connector-service.md (hunk 1 above)
Comment (mmitoraj) [reply docs/application-connector/05-04-connector-service.md]: Updated also for **deployment.args.runtimeValidityTime**.

File: docs/application-connector/05-04-connector-service.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+## Configurable parameters
+
+This table lists the configurable parameters, their descriptions, and default values:
+
+| Parameter | Description | Default value |
+|-----------|-------------|---------------|
+| **deployment.args.tokenLength**| Specifies a number of characters of a registration token. | `64` |
+| **deployment.args.appTokenExpirationMinutes** | Specifies a period of time after which a token for an Application expires. It is provided in minutes. | `5` |
+| **deployment.args.runtimeTokenExpirationMinutes** | Specifies a period of time after which a token for a runtime expires. It is provided in minutes. | `10` |
+| **deployment.args.appValidityTime** | Specifies a period of time during which certificates that the service issues for an Application are valid. It is provided in days. | `90d` |
```
Comment (kazydek) [inline docs/application-connector/05-04-connector-service.md]:
```suggestion
| **deployment.args.appValidityTime** | Specifies a period of time during which certificates that the service issues for an Application are valid. It is provided in days. | `92d` |
```

File: docs/application-connector/05-04-connector-service.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+| **deployment.args.runtimeValidityTime** | Specifies a period of time during which certificates that the service issues for a runtime are valid. It is provided in days. | `90d` |
```
Comment (kazydek) [inline docs/application-connector/05-04-connector-service.md]:
```suggestion
| **deployment.args.runtimeValidityTime** | Specifies a period of time during which certificates that the service issues for a runtime are valid. It is provided in days. | `92d` |
```

File: docs/application-connector/05-04-connector-service.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+>* [Helm overrides for Kyma installation](/root/kyma/#configuration-helm-overrides-for-kyma-installation)
+>* [Sub-charts overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation-sub-chart-overrides)
+
+## Configurable parameters
+
+This table lists the configurable parameters, their descriptions, and default values:
+
+| Parameter | Description | Default value |
+|-----------|-------------|---------------|
+| **deployment.args.tokenLength**| Specifies a number of characters of a registration token. | `64` |
```
Comment (kazydek) [inline docs/application-connector/05-04-connector-service.md]:
```suggestion
| **deployment.args.tokenLength**| Specifies a number of characters in a registration token. | `64` |
```

File: docs/application-connector/05-04-connector-service.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+| **deployment.args.appTokenExpirationMinutes** | Specifies a period of time after which a token for an Application expires. It is provided in minutes. | `5` |
+| **deployment.args.runtimeTokenExpirationMinutes** | Specifies a period of time after which a token for a runtime expires. It is provided in minutes. | `10` |
+| **deployment.args.appValidityTime** | Specifies a period of time during which certificates that the service issues for an Application are valid. It is provided in days. | `90d` |
+| **deployment.args.runtimeValidityTime** | Specifies a period of time during which certificates that the service issues for a runtime are valid. It is provided in days. | `90d` |
+| **deployment.args.central** | Determines whether the Connector Service works in the central mode. | `&central false` |
+| **deployment.envvars.country** | Specifies a country, using two-letter country codes, which is required to generate a Certificate Signing Request. | `DE` |
+| **deployment.envvars.organization** | Specifies an organization which is required to generate a Certificate Signing Request. | `Organization` |
```
Comment (kazydek) [inline docs/application-connector/05-04-connector-service.md]:
```suggestion
| **deployment.envvars.organization** | Specifies an organization which must generate a Certificate Signing Request. | `Organization` |
```

File: docs/application-connector/05-04-connector-service.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+| Parameter | Description | Default value |
+|-----------|-------------|---------------|
+| **deployment.args.tokenLength**| Specifies a number of characters of a registration token. | `64` |
+| **deployment.args.appTokenExpirationMinutes** | Specifies a period of time after which a token for an Application expires. It is provided in minutes. | `5` |
+| **deployment.args.runtimeTokenExpirationMinutes** | Specifies a period of time after which a token for a runtime expires. It is provided in minutes. | `10` |
+| **deployment.args.appValidityTime** | Specifies a period of time during which certificates that the service issues for an Application are valid. It is provided in days. | `90d` |
+| **deployment.args.runtimeValidityTime** | Specifies a period of time during which certificates that the service issues for a runtime are valid. It is provided in days. | `90d` |
+| **deployment.args.central** | Determines whether the Connector Service works in the central mode. | `&central false` |
+| **deployment.envvars.country** | Specifies a country, using two-letter country codes, which is required to generate a Certificate Signing Request. | `DE` |
```
Comment (kazydek) [inline docs/application-connector/05-04-connector-service.md]:
```suggestion
| **deployment.envvars.country** | Specifies a country which must generate a Certificate Signing Request. It is provided as a two-letter country code. | `DE` |
```

File: docs/application-connector/05-04-connector-service.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+| **deployment.envvars.organization** | Specifies an organization which is required to generate a Certificate Signing Request. | `Organization` |
+| **deployment.envvars.organizationalunit** | Specifies an organization unit which is required to generate a Certificate Signing Request. | `OrgUnit` |
```
Comment (kazydek) [inline docs/application-connector/05-04-connector-service.md]:
```suggestion
| **deployment.envvars.organizationalunit** | Specifies an organization unit which must generate a Certificate Signing Request. | `OrgUnit` |
```

Comment (kazydek) [reply docs/application-connector/05-04-connector-service.md]: and not `organizational unit` as in the parameter?

File: docs/application-connector/05-04-connector-service.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+| **deployment.envvars.locality** | Specifies a locality which is required to generate a Certificate Signing Request. | `Waldorf` |
```
Comment (kazydek) [inline docs/application-connector/05-04-connector-service.md]:
```suggestion
| **deployment.envvars.locality** | Specifies a locality which must generate a Certificate Signing Request. | `Waldorf` |
```

File: docs/application-connector/05-04-connector-service.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+| **deployment.envvars.province** | Specifies a province which is required to generate a Certificate Signing Request. | `Waldorf` |
```
Comment (kazydek) [inline docs/application-connector/05-04-connector-service.md]:
```suggestion
| **deployment.envvars.province** | Specifies a province which must generate a Certificate Signing Request. | `Waldorf` |
```

File: docs/application-connector/05-02-application-registry.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+## Configurable parameters
+
+This table lists the configurable parameters, their descriptions, and default values:
+
+| Parameter | Description | Default value |
+|-----------|-------------|---------------|
+| **deployment.args.requestTimeout** | Specifies an overall time-out after which requests to the Application Registry fail to be sent. It is provided in seconds. | `10` |
+| **deployment.args.requestLogging** | Enables logging incoming requests. By deafult, the logging is disabled. | `false` |
+| **deployment.args.specRequestTimeout** | Specifies a period of time after which a request fetching specifications provided by the user fails to be sent. It is provided in seconds. | `5` |
```
Comment (akgalwas) [inline docs/application-connector/05-02-application-registry.md]: I would use "time-out" term for consistency (as in deployment.args.requestTimeout property defined in Application Registry).

File: docs/application-connector/05-02-application-registry.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+| **deployment.args.assetstoreRequestTimeout** | Specifies a period of time after which a request fetching specifications from the Asset Store fails to be sent. It is provided in seconds. | `5` |
```
Comment (akgalwas) [inline docs/application-connector/05-02-application-registry.md]: I would use "time-out" term for consistency (as in deployment.args.requestTimeout property defined in Application Registry).

File: docs/application-connector/05-04-connector-service.md (hunk 3 above)
Comment (akgalwas) [inline docs/application-connector/05-04-connector-service.md]: I think we need to describe the correct format of the string the user can set (concerns both appValidityTime and runtimeValidityTime). The string must be of the following form: {integer value}{time unit}. Supported time units : "d", "m" and "h".

File: docs/application-connector/05-01-application-operator.md [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+>* [Helm overrides for Kyma installation](/root/kyma/#configuration-helm-overrides-for-kyma-installation)
+>* [Sub-charts overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation-sub-chart-overrides)
+
+## Configurable parameters
+
+This table lists the configurable parameters, their descriptions, and default values:
+
+| Parameter | Description | Default value |
+|-----------|-------------|---------------|
+| **controller.args.installationTimeout** | Specifies a period of time provided for Application Gateway and Event Service installation. The services are required for an Application to be operational. The value is provided in seconds.| `240` |
```
Comment (akgalwas) [inline docs/application-connector/05-01-application-operator.md]: There is also ApplicationConnectivityValidator installed. Sorry, I missed that :(. Please see the diagram: https://github.com/kyma-project/kyma/blob/master/docs/application-connector/02-01-application-connector.md

File: docs/application-connector/05-04-connector-service.md (hunk 3 above)
Comment (mmitoraj) [reply docs/application-connector/05-04-connector-service.md]: it was already changed

File: docs/application-connector/05-04-connector-service.md [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+This table lists the configurable parameters, their descriptions, and default values:
+
+| Parameter | Description | Default value |
+|-----------|-------------|---------------|
+| **deployment.args.tokenLength**| Specifies a number of characters in a registration token. | `64` |
+| **deployment.args.appTokenExpirationMinutes** | Specifies a period of time after which a token for an Application expires. It is provided in minutes. | `5` |
+| **deployment.args.runtimeTokenExpirationMinutes** | Specifies a period of time after which a token for a runtime expires. It is provided in minutes. | `10` |
+| **deployment.args.appValidityTime** | Specifies a period of time during which certificates that the service issues for an Application are valid. It is provided in days. | `92d` |
+| **deployment.args.runtimeValidityTime** | Specifies a period of time during which certificates that the service issues for a runtime are valid. It is provided in days. | `92d` |
+| **deployment.args.central** | Determines whether the Connector Service works in the central mode. | `&central false` |
```
Comment (akgalwas) [inline docs/application-connector/05-04-connector-service.md]: The default here should "false". "&central false" is a helm syntax defining reference named "central" with default value false.

File: docs/application-connector/05-04-connector-service.md (hunk 3 above)
Comment (akgalwas) [reply docs/application-connector/05-04-connector-service.md]: The default value is correct. I think we could also mention that validity time may be defined in minutes or hours, however, I can live without that.

**Pull request merged**